src/main/kotlin/com/launchdarkly/intellij/toolwindow/FlagNodeParent.kt

```diff
@@ -31,34 +23,29 @@ class FlagNodeParent(FFlag: FeatureFlag, private var flags: FeatureFlags, myProj
         return children.toTypedArray()
     }
 
+    fun updateViewModel(viewModel: FlagNodeViewModel) {
+        this.viewModel = viewModel
+    }
+
     private fun buildChildren() {
         children.add(FlagNodeBase("$KEY_PREFIX ${flag.key}", LDIcons.FLAG_KEY))
-        if (flag.description != "") children.add(FlagNodeBase("Description: ${flag.description}", LDIcons.DESCRIPTION))
+        if (viewModel.description != "") children.add(FlagNodeBase("Description: ${viewModel.description}", LDIcons.DESCRIPTION))
         children.add(FlagNodeVariations(flag))
-
-        if (env.prerequisites.isNotEmpty()) children.add(FlagNodePrerequisites(env.prerequisites, flags))
-        if (env.targets.isNotEmpty()) children.add(FlagNodeTargets(flag, env.targets))
-        if (env.rules.isNotEmpty()) children.add(FlagNodeBase("Rules: ${env.rules.size}", LDIcons.RULES))
-        if (env.fallthrough != null) children.add(FlagNodeFallthrough(flag, env))
-        if (env.offVariation != null) children.add(
-            FlagNodeBase(
-                "Off Variation: ${flag.variations[env.offVariation as Int].name ?: flag.variations[env.offVariation as Int].value}",
-                LDIcons.OFF_VARIATION
-            )
-        )
+        if (viewModel.prereqFlags.isNotEmpty()) children.add(FlagNodePrerequisites(viewModel.prereqFlags, viewModel.flags))
+        if (viewModel.targets.isNotEmpty()) children.add(FlagNodeTargets(flag, viewModel.targets))
+        if (viewModel.numRules > 0) children.add(FlagNodeBase("Rules: ${viewModel.numRules}", LDIcons.RULES))
+        if (viewModel.hasFallthrough) children.add(FlagNodeFallthrough(flag, viewModel.flagConfig!!))
+        if (viewModel.hasOffVariation) children.add(FlagNodeBase("Off Variation: ${viewModel.offVariation}", LDIcons.OFF_VARIATION))
         if (flag.tags.size > 0) children.add(FlagNodeTags(flag.tags))
     }
 
     override fun update(data: PresentationData) {
         super.update(data)
-        env = getFlags.flagConfigs[flag.key]
-            ?: FlagConfiguration(flag.key, null, null, listOf(), listOf(), arrayOf(), false, -1)
-        flag = getFlags.flags.items.find { it.key == flag.key }!!
-        // Flag version should only be -1 if we manually created a FlagConfiguration so set icon to warning.
-        val enabledIcon =
-            if (env.version === -1) LDIcons.TOGGLE_DISCONNECTED else if (env.on) LDIcons.TOGGLE_ON else LDIcons.TOGGLE_OFF
-        val label = flag.name ?: flag.key
+
+        val label = viewModel.flagLabel
         data.presentableText = label
+
+        val enabledIcon = viewModel.icon
```

Comment:
completely personal preference, so feel free to ignore, but i like grouping (setting the vars) and (updating the vals) like this

```suggestion

        val label = viewModel.flagLabel
        val enabledIcon = viewModel.icon
        
        data.presentableText = label
```

Reply:
got rid of assignment altogether since it's pretty repetitive, thanks!